src/modules/Dropdown/Dropdown.js

```diff
@@ -506,45 +600,44 @@ export default class Dropdown extends Component {
   // ----------------------------------------
 
   renderText = () => {
-    const { options, search, text } = this.props
+    const { multiple, placeholder, search, text } = this.props
     const { searchQuery, value } = this.state
+    const hasValue = multiple ? !_.isEmpty(value) : !!value
 
-    const classes = cx('text', search && searchQuery && 'filtered')
+    if (multiple) return
 
-    let _text
-    if (text) {
+    const classes = cx(
+      placeholder && !hasValue && 'default',
+      'text',
+      search && searchQuery && 'filtered'
+    )
+    let _text = placeholder
+    if (text && !searchQuery) {
       _text = text
-    } else {
-      _text = !value || searchQuery ? null : _.get(_.find(options, { value }), 'text')
+    } else if (searchQuery) {
+      _text = null
+    } else if (value) {
+      _text = _.get(this.getItemByValue(value), 'text')
     }
 
     return <div className={classes}>{_text}</div>
   }
 
-  renderPlaceholder = () => {
-    const { placeholder, search } = this.props
-    const { searchQuery, value } = this.state
-
-    if (value || !placeholder) return null
-    const classes = cx('default text', search && searchQuery && 'filtered')
-
-    return <div className={classes}>{placeholder}</div>
-  }
-
```

Comment:
Combined `text` and `placeholder`.  They are the same, the only difference is whether or not there is a `default` class applied.  This text is gray when there is a default class.  Otherwise, it is the display text of the selected value (for !multiple dropdowns).